docs/src/references/c8y-configuration-management.md

```diff
@@ -174,6 +174,47 @@ files = [
 ]
 ```
 
+Config files need not always be just plain files that can be read from or written on to the file system.
+Some applications might have non-persistent configurations which can be read/updated using some commands.
+Even for file based configurations, while updating them, just replacing the old config file with a new one may not be enough.
+You might wanna do some pre-processing steps like validating the downloaded file before applying it and
+post-processing steps like reloading the service after updating the config and so on.
+
+The `c8y-configuration-plugin` supports `extensions` to support such complex config management routines.
+Extensions can be any executable file that conforms to the following requirements:
+
+1. Support a `get` subcommand that accepts the config type as an argument and write the config for that type to `stdout`.
+   The `c8y-configuration-plugin` will capture this output to a file and upload that to the cloud.
+   In case of command failure, the exit code and `stderr` contents are sent to the cloud as failure reason.
+2. Support a `set` subcommand that accepts the config type and the downloaded config file path as arguments.
+   In case of command failure, the exit code and `stderr` contents are sent to the cloud as failure reason.
+   The `stdout` contents are ignored irrespective of success or failure (TODO: Log them as operation logs).
+
+Such extensions can be added to the `c8y-configuration-plugin.toml` file as `exts` as follows:
+
+```toml
+exts = [
+  { exec = '/etc/tedge/config-plugins/camera-agent.sh', type = 'traffic-cam-1' },
```

Comment:
I was in two-minds on specifying the `type` here in the config file. Another possible approach would have been to support a `list` command as well, just like software management plugins, that'll allow these extensions to declare the configuration types that they can support.

I stuck with this option as I felt that such extension scripts would be fairly generic and would be able to handle any number of config(app) instances. Making that script discover all the deployed configs(and even filtering them) probably is much more work for the end-user than just listing the instances that he wants to manage in the config file itself.